hazelcast/src/main/java/com/hazelcast/config/AbstractBasicConfig.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.spi.partition.IPartition;
+
+import java.io.IOException;
+
+import static com.hazelcast.util.Preconditions.checkAsyncBackupCount;
+import static com.hazelcast.util.Preconditions.checkBackupCount;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * Provides a basic configuration.
+ *
+ * @param <T> type of the config
+ */
+@SuppressWarnings("WeakerAccess")
+public abstract class AbstractBasicConfig<T extends AbstractBasicConfig> implements IdentifiedDataSerializable {
```

Comment:
what happens when on one node there is an explicit configuration, e.g. backup-count 2. And on another, which is of HZ version 3.9, the explicit configuration doesn't exist and therefor the backup count will always be 1?

Reply:
Good question, I have no idea. I just wanted to add something to the base config to give it some use within this PR. It also helped a lot to have something to test against (missed the whole dynamic config part). It also seemed to fit good into the 3.10 story to allow more than a single backup.

Reply:
I'm not quite sure but the most obvious part would be if the partition owner changed between a 3.9 and 3.8 member - one would consider having 2 replicas while the 3.8 one would consider only one replica, effectively not sending backup operations to the second replica. The other pain points might be the replication and migration checks.

Reply:
Do you mean 3.10 and 3.9? I think I could add a cluster version check and return 1 and 0 as sync/asnyc backup count, if the cluster version is < 3.10.

Could the dynamic config also be an issue? The `addAtomicLongConfig()` could try to send an `AtomicLong` config to a 3.9 member.

Reply:
Something like this should work:
```java
public abstract class AtomicLongBackupAwareOperation
    extends AbstractAtomicLongOperation
    implements BackupAwareOperation {

    @Override
    public int getSyncBackupCount() {
        ClusterService clusterService = getNodeEngine().getClusterService();
        if (clusterService.getClusterVersion().isUnknownOrLessThan(Versions.V3_10)) {
            return 1;
        }
        return getLongContainer().getConfig().getBackupCount();
    }
}
```

Reply:
Backup configuration has been removed. So this PR has no other purpose than adding an empty config class for the upcoming PRDs.